src/App.tsx

```diff
@@ -1,26 +1,246 @@
+
+  const todoInputTitleRef = useRef<HTMLInputElement>(null);
+
+  const handleAddTodoIdToloading = (todoId: Todo['id']) => {
+    setLoadingTodoIds(currentIds => [...currentIds, todoId]);
+  };
+
+  const handleRemoveTodoIdFromloading = (todoId: Todo['id']) => {
+    setLoadingTodoIds(currentIds => currentIds.filter(id => id !== todoId));
+  };
+
+  const getIsTodoLoading = (todoId: Todo['id']) =>
+    loadingTodoIds.includes(todoId);
+
+  const handleHideError = () => setErrorMessage('');
+
+  useEffect(() => {
+    if (!errorMessage) {
+      return;
+    }
+
+    const timer = setTimeout(handleHideError, 3000);
+
+    return () => clearTimeout(timer);
+  }, [errorMessage]);
+
+  useEffect(() => {
+    handleHideError();
+    async function loadTodos() {
+      try {
+        const loadedTodos = await todoServices.getTodos();
+
+        setTodos(loadedTodos);
+      } catch {
+        setErrorMessage(
+          todoErrorMessageText[TodoErrorMessage.UNABLE_TO_LOAD_TODOS],
+        );
+      }
+    }
+
+    loadTodos();
+  }, []);
+
+  const handleAddTodo = async (
+    createdNewTitle: string,
+    clearTitle: () => void,
+  ) => {
+    if (!todoInputTitleRef.current) {
+      return;
+    }
+
+    handleHideError();
+
+    // if (createdNewTitle === '') {
+    //   setErrorMessage(
+    //     todoErrorMessageText[TodoErrorMessage.TITLE_SHOULD_NOT_BE_EMPTY],
+    //   );
+    // }
+
+    const todoCreate: TodoCreate = {
+      userId: USER_ID,
+      title: createdNewTitle,
+      completed: false,
+    };
+
+    const createTepmDodo = {
+      id: 0,
+      ...todoCreate,
+    };
+
+    todoInputTitleRef.current.disabled = true;
+
+    setTempTodo(createTepmDodo);
+    try {
+      const newTodo: Todo = await todoServices.addTodo(todoCreate);
+
+      setTodos(currentTodos => [...currentTodos, newTodo]);
+      clearTitle();
+    } catch {
+      setErrorMessage(
+        todoErrorMessageText[TodoErrorMessage.UNABLE_TO_ADD_A_TODO],
+      );
+    } finally {
+      setTempTodo(null);
+      if (!todoInputTitleRef.current) {
+        return;
+      }
+
+      todoInputTitleRef.current.disabled = false;
+      todoInputTitleRef.current.focus();
+    }
+  };
+
+  const handleDeleteTodo = async (todoId: Todo['id']) => {
+    handleHideError();
+    handleAddTodoIdToloading(todoId);
+    try {
+      await todoServices.deleteTodos(todoId);
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    } catch {
+      setErrorMessage(
+        todoErrorMessageText[TodoErrorMessage.UNABLE_TO_DELETE_A_TODO],
+      );
+    } finally {
+      handleRemoveTodoIdFromloading(todoId);
+      if (!todoInputTitleRef.current) {
+        return;
+      }
+
+      todoInputTitleRef.current.focus();
+    }
+  };
+
+  const handleDeleteComplitedTodos = (todosComplited: Todo[]) => {
+    todosComplited.forEach(todoComplited => handleDeleteTodo(todoComplited.id));
+  };
+
+  const onCompletedChange = (id: number) =>
```

Comment:
```suggestion
  const handleCompletedChange = (id: number) =>
```
Be consistent with naming